Building with system GLEW library may still use bundled header files

Copying from my comment on #4391 (after I had merged it):

> Unfortunately I just found a problem with this. For most of the build, the contrib/ directory is in the include path. We have source files that include "glew/glew.h", which works for the local glew because it finds it (in contrib/glew/glew.h). However, the system header is /usr/include/GL/glew.h. That obviously won't be found at "glew/glew.h"; it would be found at <GL/glew.h>.
>
> So I think with this patch and when configured to use system glew this will actually link against the system library but use the header from the contrib/ library. That will cause a version skew problem except for people who are "lucky" enough that their installed system glew version exactly matches the one we've bundled.
>
> We'll need to move the headers around in contrib/ so that the include structure is more uniform and then conditionally add contrib/glew (or whatever) to the include path when we're using the bundled copy. Or some other fix.

This is what I get now:

 CMake Error at /usr/share/cmake-3.12/Modules/FindGLEW.cmake:50 (set_target_properties):
   set_target_properties can not be used on an ALIAS target.
 Call Stack (most recent call first):
   CMakeLists.txt:88 (find_package)

CMake Error at /usr/share/cmake-3.12/Modules/FindGLEW.cmake:50 
    (set_target_properties):
    set_target_properties can not be used on an ALIAS target.
    Call Stack (most recent call first):
    CMakeLists.txt:88 (find_package)

The ALIAS is only created in the `if (NOT USE_SYSTEM_LIBGLEW)` branch, and find_package(GLEW) is only called in the `if (USE_SYSTEM_LIBGLEW)` branch. I'm not sure how they could end up both running and conflicting like this.

Perhaps something weird happens if cmake configure is run with one config and then run again with a different config? cmake does a bunch of variable caching... maybe it's caching some information about targets.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,6 +85,7 @@ endif (WIN32)
 
 option(USE_SYSTEM_LIBGLEW "Use the system's libglew" OFF)
 if (USE_SYSTEM_LIBGLEW)
+	add_library(GLEW::GLEW INTERFACE IMPORTED)
 	find_package(GLEW REQUIRED)
 endif (USE_SYSTEM_LIBGLEW)
 

